x-pack/plugin/ilm/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ilm/CCRIndexLifecycleIT.java

```diff
@@ -746,10 +746,19 @@ private static Object getIndexSetting(RestClient client, String index, String se
         return settings.get(setting);
     }
 
-    private static void assertDocumentExists(RestClient client, String index, String id) throws IOException {
+    private void assertDocumentExists(RestClient client, String index, String id) throws IOException {
         Request request = new Request("HEAD", "/" + index + "/_doc/" + id);
-        Response response = client.performRequest(request);
-        assertThat(response.getStatusLine().getStatusCode(), equalTo(200));
+        Response response;
+        try {
+            response = client.performRequest(request);
+            if (response.getStatusLine().getStatusCode() != 200) {
+                logger.error(EntityUtils.toString(response.getEntity()));
+                fail("HTTP response code expected to be [200] but was [" + response.getStatusLine().getStatusCode() + "]");
+            }
+        } catch (ResponseException ex) {
+            logger.error(EntityUtils.toString(ex.getResponse().getEntity()));
```

Comment:
Maybe also include the stacktrace just in case?

```suggestion
            logger.error(EntityUtils.toString(ex.getResponse().getEntity()), ex);
```

Reply:
Good thought! Done.